platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -366,4 +423,68 @@ private int findJoystickDevice(int device_id) {
 
 		return -1;
 	}
+
+	private boolean handleMouseEvent(final MotionEvent event) {
+		switch (event.getActionMasked()) {
+			case MotionEvent.ACTION_HOVER_ENTER:
+			case MotionEvent.ACTION_HOVER_MOVE:
+			case MotionEvent.ACTION_HOVER_EXIT: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int type = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.hover(type, x, y);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_BUTTON_PRESS:
+			case MotionEvent.ACTION_BUTTON_RELEASE:
+			case MotionEvent.ACTION_MOVE: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int buttonsMask = InputUtils.fixButtonsStateMask(event.getButtonState());
+				final int action = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.touch(event.getSource(), action, 0, 1, new float[] { 0, x, y }, buttonsMask, 1);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_SCROLL: {
+				final float[] positionArray = new float[] { 0, event.getX(), event.getY() };
+				final int buttonsMaskFixed = InputUtils.fixButtonsStateMask(event.getButtonState());
+
+				if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) > 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_UP, event.getAxisValue(MotionEvent.AXIS_VSCROLL), positionArray);
+				} else if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) < 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_DOWN, -event.getAxisValue(MotionEvent.AXIS_VSCROLL), positionArray);
+				}
+
+				if (event.getAxisValue(MotionEvent.AXIS_HSCROLL) > 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_RIGHT, event.getAxisValue(MotionEvent.AXIS_HSCROLL), positionArray);
+				} else if (event.getAxisValue(MotionEvent.AXIS_HSCROLL) < 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_LEFT, -event.getAxisValue(MotionEvent.AXIS_HSCROLL), positionArray);
+				}
+				return true;
+			}
+		}
+		return false;
+	}
+
+	private void handleMouseScrollWheelButtonEvent(final int source, final int buttonsMaskFixed, final int scrollButton, final float factor, final float[] positionArray) {
+		queueEvent(new Runnable() {
+			@Override
+			public void run() {
+				int fixedButtonsMaskWithScrollButton = buttonsMaskFixed | scrollButton;
```

Comment:
If I understand correctly, this is mixing a button mask from Android with constants from Godot? It needs to be done in a way that avoids possible overlaps or side effects, since these masks are not directly related.

Reply:
As I mentioned above godot's and android's buttonmasks are pretty close, and after fixing it's only godot's buttonmasks